src/persist/src/indexed/mod.rs

```diff
@@ -1401,19 +1401,57 @@ mod tests {
         // Different key codec
         assert_eq!(
             block_on(|res| i.register("stream", "nope", "val", res)),
-            Err(IndexedError::from(
+            Err(Error::from(
                 "invalid registration: key codec mismatch nope vs previous key"
             ))
         );
 
         // Different val codec
         assert_eq!(
             block_on(|res| i.register("stream", "key", "nope", res)),
-            Err(IndexedError::from(
+            Err(Error::from(
                 "invalid registration: val codec mismatch nope vs previous val"
             ))
         );
 
         Ok(())
     }
+
+    #[test]
+    fn regression_8303_snapshot_forward_since() -> Result<(), Error> {
+        let mut i = MemRegistry::new().indexed_no_reentrance()?;
+        let id = block_on(|res| i.register("0", "", "", res))?;
+
+        // Introduce some data, seal it, forward since. Intentionally don't call
+        // step so that we don't have a chance to compact it (which did produce
```

Comment:
I'm a little bit confused by the text in parentheses - is that saying "most of the time in the past we did get around to compacting everything and therefore didn't see the bug" or something else?